builtin/common/misc_helpers.lua

```diff
@@ -345,41 +345,20 @@ if INIT == "game" then
 		end
 		local undef = core.registered_nodes[unode.name]
 		if undef and undef.on_rightclick then
-			undef.on_rightclick(pointed_thing.under, unode, placer,
+			return undef.on_rightclick(pointed_thing.under, unode, placer,
```

Comment:
Note to reviewers: `on_rightclick` may return an itemstack too, depending on what the player shall hold after clicking the node. Another minor bugfix.